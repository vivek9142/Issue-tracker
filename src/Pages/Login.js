import React,{useEffect,useState} from 'react';
import {Formik,Form} from 'formik';
import TextField from '../Components/TextField';
import * as Yup from 'yup';

import { useDispatch,useSelector } from 'react-redux';
import { userActions } from '../Redux/Reducer/userReducer';

const Login = (props) => {
    const auth = useSelector(state => state.user.isAuthenticated);
    const [err,setErr] = useState('');
    console.log(auth);
    useEffect(() => {
        if(auth)
        props.history.push('/');
        
        return () => {
        }
    }, [props.history,auth]);

    const  dispatch = useDispatch();
    
    const validate = Yup.object({
        email:Yup.string()
        .email('Email is invalid')
        .required('Email is Required'),
        password:Yup.string()
        .min(6,'Password must be atleast 6 charaters long')
        .required('Password is Required')
    })

    const loginHandler = (user) => {
        
        dispatch(userActions.loginUser(user));
        if(!user.isAuthenticated) setErr('Wrong Username or Password');
        
    }
    
    return (
        <>
        <h1>Login Page</h1>
        <Formik initialValues={{
            email:''
            ,password:''
        }} 
        validationSchema={validate} 
        onSubmit={val => loginHandler(val)}>

            {formik => (
                <Form>
                    <span>{ err }</span>
                    <TextField type='email' name='email' label='Email'/>
                    <TextField type='password' name='password' label='Password'/>
                    <input type='submit'/>
                </Form>
            )}
        </Formik>
        </>
    );
}

export default Login;