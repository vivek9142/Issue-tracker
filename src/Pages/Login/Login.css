.user-section{
    width: 70%;
}
.user-section--auth-container{
    max-width: 50%;
}
@media screen and (max-width: 992px) {
    .user-section{
        width: 75%;
    }
    .user-section--auth-container{
        max-width: 60%;
    }
}

@media screen and (max-width: 582px) {
    .user-section{
        width: 100%;
    }
    .user-section--auth-container{
        max-width: 95%;
    }
}