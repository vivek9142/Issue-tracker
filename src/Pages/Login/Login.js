import React,{useEffect} from 'react';
import {Formik,Form} from 'formik';
import TextField from '../../Components/TextField/TextField';
import * as Yup from 'yup';

import { useDispatch,useSelector } from 'react-redux';
import { loginUser } from '../../Redux/Actions/UserActions';
import { userActions } from '../../Redux/Reducer/userReducer';
import './Login.css';

const Login = (props) => {
    
    const  dispatch = useDispatch();

    useEffect(() => {
        return () => {
            dispatch(userActions.resetErrorStatus());
        }
    },[dispatch]);
    
    const authError = useSelector(state => state.user.error);
    const validate = Yup.object({
        email:Yup.string()
        .email('Email is invalid')
        .required('Email is Required'),
        password:Yup.string()
        .min(6,'Password must be atleast 6 charaters long')
        .required('Password is Required')
    })

    const loginHandler = (user) => dispatch(loginUser(user));
    
    return (
        <>
        <div className="container user-section mt-5 mb-3 ">
            <h3 className="text-center">
                Login
            </h3>
            <hr/>
        </div>
        <div className="container mb-5">
            <h5 className="fs-6 text-center fw-normal ">{authError}
            </h5>
        </div>

        <Formik initialValues={{
            email:''
            ,password:''
        }} 
        validationSchema={validate} 
        onSubmit={val => loginHandler(val)}>

            {formik => (
                <>
                <div className="container user-section--auth-container">
                    <Form>
                    <div className="row g-3">
                            
                            <div className="col-12">
                            <div className="">
                                <label htmlFor='email' className='form-label'>
                                    Email</label>
                                    <input type='email' name='email' className='form-control' placeholder={`Enter your Email`}/>
                                    <div style={{"height":"1rem"}}>
                                    <span name='email' style={{"fontSize":"0.8rem","color":"red"}}/>
                                    </div>
                            </div>
                                
                            </div>
                            <div className="col-12">
                                <TextField type='password' name='password' label='Password'/>
                            </div>

                            <div className="col-12" style={{'marginBottom':'5rem'}}>
                                <button type="submit" className="btn btn-primary">Sign in</button>
                            </div>
                    </div>
                    </Form>
                </div>
                </>
            )}
        </Formik>
        </>
    );
}

export default Login;