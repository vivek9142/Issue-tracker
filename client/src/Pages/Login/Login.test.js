import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Login from './Login';
import {Provider} from 'react-redux';
import store from '../../Redux/Store';
import { MemoryRouter } from 'react-router-dom';

describe('Login page Page Test Suite', () => {
    beforeEach(()=>{
        
    render(
        <MemoryRouter>   
        <Provider store={store}><Login/></Provider>
        </MemoryRouter> 
        );
    });
    test('Login page Snapshot',()=>{
        const tree = renderer.create(<MemoryRouter>   
            <Provider store={store}><Login/></Provider>
            </MemoryRouter> );
        expect(tree).toMatchSnapshot();
      });
      
    test('renders main Heading of Login page Page', () => {
        const element = screen.getByText(/Login/i);
        expect(element).toBeInTheDocument();
    });

    test('Login page should have inputs', () => {
        expect(screen.getAllByRole('textbox')).toHaveLength(1);
        expect(screen.getByText(/Login/i)).toBeInTheDocument()
    });

    test('Login page should have a submit button', () => {
        const submitButton = screen.getByRole('button', { name: 'Sign in' });
        expect(submitButton).toBeDefined()
    });

  
})