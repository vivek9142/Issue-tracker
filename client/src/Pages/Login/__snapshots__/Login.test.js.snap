// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login page Page Test Suite Login page Snapshot 1`] = `
Array [
  <div
    className="container user-section mt-5 mb-3 "
  >
    <h3
      className="text-center"
    >
      Login
    </h3>
    <hr />
  </div>,
  <div
    className="container mb-5"
  >
    <h5
      className="fs-6 text-center fw-normal "
    >
      
    </h5>
  </div>,
  <div
    className="container user-section--auth-container"
  >
    <form
      action="#"
      onReset={[Function]}
      onSubmit={[Function]}
    >
      <div
        className="row g-3"
      >
        <div
          className="col-12"
        >
          <div
            className=""
          >
            <label
              className="form-label"
              htmlFor="Email"
            >
              Email
            </label>
            <input
              className="form-control"
              name="email"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Enter your Email"
              type="email"
              value=""
            />
            <div
              style={
                Object {
                  "height": "1rem",
                }
              }
            />
          </div>
        </div>
        <div
          className="col-12"
        >
          <div
            className=""
          >
            <label
              className="form-label"
              htmlFor="Password"
            >
              Password
            </label>
            <input
              className="form-control"
              name="password"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Enter your Password"
              type="password"
              value=""
            />
            <div
              style={
                Object {
                  "height": "1rem",
                }
              }
            />
          </div>
        </div>
        <div
          className="col-12"
          style={
            Object {
              "marginBottom": "5rem",
            }
          }
        >
          <button
            className="btn btn-primary"
            type="submit"
          >
            Sign in
          </button>
        </div>
      </div>
    </form>
  </div>,
]
`;
